pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -158,3 +189,55 @@ func (nc *nodeConfig) handleCSR(requestorFilter string) error {
 
 	return nil
 }
+
+// GetNode returns a pointer to the node object associated with the external IP provided
+func (nc *nodeConfig) getNode(externalIP string) (*v1.Node, error) {
+	var matchedNode *v1.Node
+
+	nodes, err := nc.k8sclientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not get list of nodes")
+	}
+	if len(nodes.Items) == 0 {
+		return nil, fmt.Errorf("no nodes found")
+	}
+
+	// Find the node that has the given IP
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type == "ExternalIP" && address.Address == externalIP {
+				matchedNode = &node
```

Comment:
We may have to change this to get the node object by `instanceID` from cloud provider method. This is to ensure that say in future if we don't rely on public ip to get the address. Please look at this [method](https://github.com/ravisantoshgudimetla/windows-machine-config-operator-1/commit/e6a5579b6527dcadb9d9d8471c7cabf7306405a8#diff-6a80c7c59a5c0c59d705c805a83830e3R90)

Reply:
Yes or else whoever is picking that up can handle it.

Reply:
@PratikMahajan are you okay with addressing this in your PR?

Reply:
i'll pick this up!